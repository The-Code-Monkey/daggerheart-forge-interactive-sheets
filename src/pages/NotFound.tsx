import { useLocation } from "react-router-dom";
import { JSX, useEffect } from "react";

const NotFound = (): JSX.Element => {
  const location = useLocation();

  useEffect(() => {
    console.log(
      "404 Error: User attempted to access non-existent route:",
      location.pathname
    );
  }, [location.pathname]);

  return (
    <div className="min-h-screen flex items-center justify-center bg-linear-to-b from-brand-900 via-brand-800 to-slate-900">
      <div className="text-center">
        <h1 className="text-4xl font-bold mb-4 text-white">404</h1>
        <p className="text-xl text-brand-200 mb-4">Oops! Page not found</p>
        <a
          href="/"
          className="text-yellow-400 hover:text-yellow-300 underline font-semibold"
        >
          Return to Home
        </a>
      </div>
    </div>
  );
};

export default NotFound;
