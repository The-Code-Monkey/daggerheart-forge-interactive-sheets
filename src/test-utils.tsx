import React from "react";
import { render, RenderOptions } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

// Mock AuthContext
const mockAuthContext = {
  user: null,
  loading: false,
  signIn: jest.fn(),
  signOut: jest.fn(),
  signUp: jest.fn(),
};

const AuthProvider = ({ children }: { children: React.ReactNode }) => {
  return <div data-testid="auth-provider">{children}</div>;
};

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <BrowserRouter>
      <AuthProvider>{children}</AuthProvider>
    </BrowserRouter>
  );
};

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

// Export everything from testing-library/react
export * from "@testing-library/react";
export { customRender as render };

// Mock useAuth hook
export const mockUseAuth = () => mockAuthContext;
