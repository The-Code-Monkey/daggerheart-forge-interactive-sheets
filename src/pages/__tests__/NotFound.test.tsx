
import { render } from "../../test-utils";
import NotFound from "../NotFound";

// Mock react-router-dom
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: () => ({
    pathname: "/nonexistent",
  }),
}));

describe("NotFound Page", () => {
  it("renders correctly", () => {
    const { container } = render(<NotFound />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
