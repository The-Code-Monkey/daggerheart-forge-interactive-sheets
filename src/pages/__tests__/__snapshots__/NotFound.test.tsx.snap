// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Page renders correctly 1`] = `
<div
  data-testid="auth-provider"
>
  <div
    class="min-h-screen flex items-center justify-center bg-gray-100"
  >
    <div
      class="text-center"
    >
      <h1
        class="text-4xl font-bold mb-4"
      >
        404
      </h1>
      <p
        class="text-xl text-gray-600 mb-4"
      >
        Oops! Page not found
      </p>
      <a
        class="text-blue-500 hover:text-blue-700 underline"
        href="/"
      >
        Return to Home
      </a>
    </div>
  </div>
</div>
`;
